diff --git a/src/Data/GADT/Compare/TH.hs b/src/Data/GADT/Compare/TH.hs
index 3dbfead..6f35220 100644
--- a/src/Data/GADT/Compare/TH.hs
+++ b/src/Data/GADT/Compare/TH.hs
@@ -89,6 +89,9 @@ geqClause bndrs con = do
 newtype GComparing a b t = GComparing (Either (GOrdering a b) t)
 
 instance Functor (GComparing a b) where fmap f (GComparing x) = GComparing (either Left (Right . f) x)
+instance Applicative (GComparing a b) where
+    pure = return
+    (<*>) = ap
 instance Monad (GComparing a b) where
     return = GComparing . Right
     GComparing (Left  x) >>= f = GComparing (Left x)
